src/index.js

```diff
@@ -0,0 +1,43 @@
+import currentLocation from './currentLocation';
+import render from './render';
+
+const searchBar = document.querySelector('.search-bar');
+const locationIcon = document.querySelector('.location-icon');
+
+const weather = (long, lat) => fetch(`http://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${long}&units=metric&APPID=4e1f41a544e8138e1382d0bff94cc988`, { mode: 'cors' })
```

Comment:
Let us keep data fetching separate from the `render` of the data. We can also use `async` and `await` here:

```javascript
const weather = async (long, lat) => {
  const url = `http://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${long}&units=metric&APPID=4e1f41a544e8138e1382d0bff94cc988`;
  const result = await fetch(url, {mode: 'cors'})
                            .then(response => {
                            if(response.status === 200) return response.json();
                                             else return Promise.reject("Cannot fetch data");
                              });
   return result;
} 
```

We can then use the result of `weather` in the downstream processes by chaining `then` and `catch` statements.